TM1py/Utils/Utils.py

```diff
@@ -1321,3 +1322,59 @@ def init_poolmanager(self, *args, **kwargs):
         if self.socket_options is not None:
             kwargs["socket_options"] = self.socket_options
         super(HTTPAdapterWithSocketOptions, self).init_poolmanager(*args, **kwargs)
+
+def sync_two_dimensional_cube(tm1_s: TM1Service, tm1_t: TM1Service, cube_name: str,
+                                  skip_non_existing_elements=False, sync_rules=False):
```

Comment:
please add one additional arguments to this function: `max_workers`.

The argument can simply be delegated to the `execute_mdx` to the multithread the retrieval.

on the other side you need to use `write_async` if max_workers is greater than 1.